src/blog/2019/002-bpmn-js-3-4-0.md

```diff
@@ -0,0 +1,113 @@
+---
+
+title: Improved Connection Layout And Multiple BPMN Diagrams Support
+layout: blogpost.hbs
+
+slug: 2019-bpmn-js-3-4-0
+author: Maciej Barelkowski<https://github.com/barmac>
+
+published: 2019-04-26 10:00
+
+releases:
+  - 'bpmn-js@3.4.0'
+  - 'diagram-js@3.3.0'
+
+---
+
+<p class="introduction">
+  We're proud to announce the latest release of our [BPMN modeling toolkit](https://github.com/bpmn-io/bpmn-js). It ships with several improvements regarding connections layout as well as important modeling fixes and support for multiple BPMN diagrams within one file.
+</p>
+
+<!-- continue -->
+
+The connections layout in [bpmn-js](https://github.com/bpmn-io/bpmn-js) has already been greatly improved in [bpmn-js v2.4.0 release](https://bpmn.io/blog/posts/2018-bpmn-js-2-4-0.html). However, it still caused issues when a boundary event was connected to a flow element straight below or when reconnecting elements. It is not the case anymore.
+
+<div class="figure">
+  <img src="{{ assets }}/attachments/blog/2019/002-boundary-layout.gif" alt="Connection from a boundary event to a flow element is correctly laid out.">
+  <p class="caption">
+    The connection is perfectly laid out when an element is moved straight below the boundary event.
+  </p>
+</div>
+
+Numerous layout improvements introduced in this release appropriately handle these cases.
+
+## Improved Reconnecting Layout
+
+Moving a connection start or end is now handled adequately.
+
+<div class="figure">
+  <img src="{{ assets }}/attachments/blog/2019/002-reconnect-end-layout.gif" alt="Connection is laid out after its end is moved.">
+  <p class="caption">
+    Reconnecting elements looks better than ever.
+  </p>
+</div>
+
+As a side-effect, the new layout behavior is applied also after element removal. Element's connections will be joined and adequately laid out.
+
+<div class="figure">
+  <img src="{{ assets }}/attachments/blog/2019/002-boundary-removal.gif" alt="Properly laid out connection after element removal.">
+  <p class="caption">
+    Connection is now properly laid out according to Manhattan Layout.
+  </p>
+</div>
+
+## Collapsed Pool Enhancement
+
+In the current release, the pool shrinks automatically after the collapse. You don't need to do it manually anymore.
+
+<div class="figure">
+  <img src="{{ assets }}/attachments/blog/2019/002-participant-collapse.gif" alt="A participant is automatically resized after collapse.">
+  <p class="caption">
+    Collapsing a pool will make it as flat as a pancake.
+  </p>
+</div>
+
+The height of the pool changes to the default value, but the width is left as it was.
```

Comment:
Is this line necessary? I think it is going to be obvious for users, right?

Reply:
Yeah, I could as well remove it.